### PR TITLE
Optimizing sqlType actuator based on static factory pattern

### PR DESCRIPTION
**Pull request by pengzhengfa**

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java [hunk 1]
```diff
@@ -78,23 +80,7 @@
...
-                    executor = new DeleteExecutor<>(statementProxy, statementCallback, sqlRecognizer);
-                    break;
-                case SELECT_FOR_UPDATE:
-                    executor = new SelectForUpdateExecutor<>(statementProxy, statementCallback, sqlRecognizer);
-                    break;
-                default:
-                    executor = new PlainExecutor<>(statementProxy, statementCallback);
-                    break;
-            }
+            executor = createSqlExecutor(sqlRecognizer.getSQLType(), sqlRecognizer, statementProxy, statementCallback);
```
Comment (l81893521) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java]: I think
```
createSqlExecutor(sqlRecognizer, statementProxy, statementCallback)
```
that is enough.

Comment (pengzhengfa) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java]: @l81893521 Thank you for your advice

Comment (pengzhengfa) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java]: @l81893521 Createsqlexecutor (sqlrecognizer, statementproxy, statementcallback) needs to return parameters to execute SQL for the executor

Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java:83]: I m apologize that I did not clear my point. I mean 3 parameters that s enough. just like that below.
```
executor = createSqlExecutor(sqlRecognizer, statementProxy, statementCallback);
```

Comment (pengzhengfa) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java:83]: @l81893521 OK, I'll change it

**Pull request closed without merging**